server/src/gnome/secret_exchange.rs

```diff
@@ -0,0 +1,174 @@
+// SecretExchange: Exchange secrets between processes in an unexposed way.
+
+// Initial C implementation: https://gitlab.gnome.org/GNOME/gcr/-/blob/master/gcr/gcr-secret-exchange.c
+
+// The initial implementation of SecretExchange/GCRSecretExchange uses a KeyFile
+// to encode/parse the payload. In this implementation the payload is based
+// on a HashMap.
+// Before any transit operations the payload is base64 encoded and parsed into a
+// String.
+
+use std::collections::HashMap;
+
+use base64::prelude::*;
+use oo7::{crypto, Key};
+use zeroize::Zeroize;
+
+const SECRET: &str = "secret";
+const PUBLIC: &str = "public";
+const PRIVATE: &str = "private";
+const IV: &str = "iv";
+const PROTOCOL: &str = "[sx-aes-1]\n";
+const CIPHER_TEXT_LEN: usize = 16;
+
+#[derive(Debug, Zeroize)]
+pub struct SecretExchange {
```

Comment:
After giving this some thoughts and looking at the current usage in the prompting code. I would drop this struct and also drop the begin method. Instead, inside PrompterCallback you would store the `Key`s directly.

Then have the following methods:
- `secret_exchange::begin(public_key: &Key)`
- `secret_exchange::handshake(client_private_key: &Key, server_public_key: &Key)`
- `secret_exchange::retrieve(exchange: &str, aes_key: &Key)` 

As a first step, a second step would consists of creating a wrapper type for the Exchange that holds a String internally and have an easy way to convert From/To a HashMap to simplify the serialization/deserialization.